Continue updating libpacemaker to current guidelines

kgaillot opened the pull request:
This continues the "best practices" project for libpacemaker. Earlier PRs addressed most constraint-related functions, and this is a follow-up to clean up the remaining such functions -- mostly renaming and moving to appropriate source files.

kgaillot commented:
@clumens , can you review this when you have the chance? Low priority

clumens commented on lib/pacemaker/pcmk_sched_colocation.c:
```diff
@@ -799,3 +799,113 @@ pcmk__block_colocated_starts(pe_action_t *action, pe_working_set_t *data_set)
+
+/*!
+ * \internal
+ * \brief Determine what colocation constraint should affect at this point
```
Should this say `Determine what colocation constraint should be in effect at this point`?  Or something else?  It seems to be missing a word as written.

clumens replied:
Oh I see, constraint is a parameter.  Can you make it `\p constraint` or `the constraint parameter` or something?

kgaillot replied:
done

kgaillot commented on include/pcmki/pcmki_sched_allocate.h:
```diff
@@ -30,6 +30,26 @@ struct resource_alloc_functions_s {
+     * \note This function is recursive; top-level callers should pass NULL as
+     *       \p colocated_rscs and \p orig_rsc, and the desired resource as
+     *       \p rsc. The recursive calls will use other values.
+     */
```
Yep I wanted to avoid that originally too :) but this is a function pointer in a struct, with multiple implementations. We could wrap it to avoid calling it directly, but the function pointer itself would still need the recursive signature:

    static inline GList *
    pcmk__something(pe_resource_t *rsc)
    {
        return rsc->cmds->colocated_resources(rsc, NULL, NULL);
    }

kgaillot commented on lib/pacemaker/pcmk_sched_group.c:
```diff
@@ -599,3 +599,44 @@ void
+    }
+
+    if (group_data->colocated || pe_rsc_is_clone(rsc->parent)) {
+        /* This is group has colocated members and/or is cloned -- either way,
```
done

kgaillot commented:
Updated per review and added one huge commit on top to address a long-time annoyance. @clumens sorry for the extra commit to review but I couldn't resist ;)

clumens commented:
This looks good to me.
